File: easy/two_sum/two_sum.go
package two_sum

// https://leetcode.com/problems/two-sum
// https://leetcode.com/submissions/detail/736602720/
func TwoSum(nums []int, target int) []int {
	m := make(map[int]int, len(nums))

	for i := range nums {
		j, exists := m[nums[i]]
		if exists {
			return []int{j, i}
		}
		m[target-nums[i]] = i
	}

	return []int{}
}

File: top_150/easy/solutions/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d = dict()
        for i in range(len(nums)):
            ival = nums[i]
            if ival in d:
                return [d.get(ival), i]
            d[target-ival] = i
